lib/web/purchases_flutter_web.dart

```diff
@@ -0,0 +1,440 @@
+import 'dart:async';
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:web/web.dart';
+
+import '../purchases_flutter.dart';
+
+class PurchasesFlutterPlugin {
+  static final _unknownErrorCode = '${PurchasesErrorCode.unknownError.index}';
+  static final _configurationErrorCode = '${PurchasesErrorCode.configurationError.index}';
+  static const _platformName = 'flutter';
+  static const _pluginVersion = '8.7.3';
+
+  static Completer<void>? _initCompleter;
+
+  static void _injectScriptIfNeeded() {
+    if (_initCompleter != null) {
+      return;
+    }
+    _initCompleter = Completer<void>();
+
+    // TODO: Host library ourselves.
+    const library = 'https://unpkg.com/@revenuecat/purchases-js-hybrid-mappings@0.0.7-alpha.2/dist/index.umd.js';
+
+    final script = HTMLScriptElement()
+      ..type = 'text/javascript'
+      ..text = '''
+        window.after_rc_load_callback = async (callback) => {
+          callback(await import("$library"));
+        };
+      ''';
+
+    assert(document.head != null);
+    document.head!.append(script);
```

Comment:
`assert` followed by bang operator (postfix null assertion; `!`) would _in theory_ result in different debug and release mode errors (`_AssertionError` vs `_TypeError`). I would probably check for nullability and explicitly throw a relevant error, e.g.:

```suggestion
    final head = document.head;
    if (head == null) {
      throw StateError(
        "The current document doesn't have a head element which is required to insert a script.",
      );
    }
    
    head.append(script);
```

---
_that being said_, even if a developer removes the `head` section, Flutter would add it back automatically, so I guess it's nullable only to be compatible with the Web IDL standard 🤷‍♂️

Reply:
That makes sense. Thanks for the suggestion!